skimage/feature/haar.py

```diff
@@ -0,0 +1,319 @@
+from __future__ import division
+
+from itertools import chain
+from operator import add
+
+import six
+import numpy as np
+
+from ._haar import haar_like_feature_coord_wrapper
+from ._haar import haar_like_feature_wrapper
+from ..color import gray2rgb
+from ..draw import rectangle
+from ..exposure import rescale_intensity
+from .._shared.utils import check_random_state
+from ..util import img_as_float
+
+FEATURE_TYPE = ('type-2-x', 'type-2-y',
+                'type-3-x', 'type-3-y',
+                'type-4')
+
+
+def _validate_feature_type(feature_type):
+    """Transform feature type to an iterable and check that it exists."""
+    if feature_type is None:
+        feature_type_ = FEATURE_TYPE
+    else:
+        if isinstance(feature_type, six.string_types):
+            feature_type_ = [feature_type]
+        else:
+            feature_type_ = feature_type
+        for feat_t in feature_type_:
+            if feat_t not in FEATURE_TYPE:
+                raise ValueError('The given feature type is unknown. Got {}'
```

Comment:
Can newline after opening bracket, then you have a bit more room to work with here